WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/GameFiles/LevelManager.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class LevelManager : MonoBehaviour {
+
+    public string overworldSceneName;
+    public string swimmingGameSceneName;
+
+    public static LevelManager Instance;
+
+    void Awake()
+    {
+        //Check if instance already exists
+        if (Instance == null)
+            Instance = this;
+        else if (Instance != this)
+            Destroy(this);
```

Comment:
This Awake() method is black magic to me. Why is this necessary? What does this do?

Reply:
Making the class a singleton. If level manager ever gets accidentally instantiated a second time, it will just destroy itself so it doesn't mess anything up.